### PR TITLE
Update normal contact force for frictional node-face penalty 

### LINKED ISSUE #24632: Unphysical oscillations in tangential forces for contact problem
Look into formulation for penalty node-face contact, particularly the definition of tangential traction from the normal pressure and its Jacobian. Look at one benchmark problem.

See discussion below.

### PR DESCRIPTION
**Pull request by recuero**
Closes #24632.

### CODE REVIEW

File: modules/contact/test/tests/sliding_block/sliding/tests [hunk 1]
```diff
@@ -25,11 +25,12 @@
     abs_zero = 1e-7
     rel_err = 1e-4
     max_time = 800
-#    deleted = '#6139'
+    allow_warnings = true
+    #    deleted = '#6139'
```
Comment (lindsayad) [inline modules/contact/test/tests/sliding_block/sliding/tests:29]:
```suggestion
```

File: modules/contact/test/tests/sliding_block/sliding/tests [hunk 2]
```diff
@@ -38,11 +39,11 @@
     abs_zero = 1e-7
     rel_err = 1e-4
     max_time = 800
-#    deleted = '#6139'
+    #    deleted = '#6139'
```
Comment (lindsayad) [inline modules/contact/test/tests/sliding_block/sliding/tests:42]:
```suggestion
```

File: modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    type = PiecewiseLinear
+    x = '0. 1. 3.5'
+    y = '0. 0.0 0.015'
+  []
+[]
+
+[Kernels]
+  [TensorMechanics]
+    use_displaced_mesh = true
+    save_in = 'saved_x saved_y'
```
Comment (lindsayad) [inline modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i:75]: Do we need the save in stuff?

File: modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    family = MONOMIAL
+  []
+  [saved_x]
+  []
+  [saved_y]
+  []
+  [diag_saved_x]
+  []
+  [diag_saved_y]
+  []
```
Comment (lindsayad) [inline modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i:46]: Do we need the save-in stuff?

File: modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i (hunk 3 above)
Comment (recuero) [reply modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i]: No, we don't need it anymore.

**Review by lindsayad — APPROVED** [review]

**Review by bwspenc — APPROVED** [review]
Looks good.

**Pull request merged**